### PR TITLE
More TextField example updates

### PR DESCRIPTION
**Pull request by ecraig12345**
#### Pull request checklist

- [x] Include a change request file using `$ npm run change`

#### Description of changes

- Add controlled examples
- Remove the number example since it uses bad practices and generally wasn't adding much value
- Lay out some examples in two columns

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/8540)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/TextField/examples/TextField.Controlled.Example.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+  private _onChange1 = (ev: React.FormEvent<HTMLInputElement>, newValue?: string) => {
+    this.setState({ value1: newValue || '' });
+  };
+
+  private _onChange2 = (ev: React.FormEvent<HTMLInputElement>, newValue?: string) => {
+    if (!newValue || newValue.length <= 5) {
+      this.setState({ value2: newValue || '' });
+    } else {
+      // This block should NOT be necessary, but there's currently a bug (#1350) where a controlled
```
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/TextField/examples/TextField.Controlled.Example.tsx]: This is the major downside to adding a controlled example right now...the TextField doesn't actually behave right when controlled, and fixing that would be such a major behavior change that I'm hesitant to do it outside a major version bump. So there's this extra block and giant comment needed to get the right behavior.

File: packages/office-ui-fabric-react/src/components/TextField/TextField.doc.tsx [hunk 2]
```diff
@@ -6,12 +6,10 @@ import { TextFieldBorderlessExample } from './examples/TextField.Borderless.Exam
 import { TextFieldCustomRenderExample } from './examples/TextField.CustomRender.Example';
 import { TextFieldErrorMessageExample } from './examples/TextField.ErrorMessage.Example';
 import { TextFieldMultilineExample } from './examples/TextField.Multiline.Example';
-import { TextFieldPlaceholderExample } from './examples/TextField.Placeholder.Example';
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/TextField/TextField.doc.tsx]: We were using the placeholder examples when verifying the fluent styles for placeholder text. Can we include them back maybe within the basic examples?

Comment (ecraig12345) [reply packages/office-ui-fabric-react/src/components/TextField/TextField.doc.tsx]: I combined the placeholder examples and basic examples.

File: packages/office-ui-fabric-react/src/components/TextField/examples/TextField.Basic.Example.tsx [hunk 3]
```diff
@@ -3,15 +3,25 @@ import { TextField, MaskedTextField } from 'office-ui-fabric-react/lib/TextField
...
+        <TextField label="Standard" />
+        <TextField label="Disabled" disabled defaultValue="I am disabled" />
+        <TextField label="Read-only" readOnly defaultValue="I am read-only" />
+        <TextField label="Required " required />
+        <TextField label="With error message" errorMessage="Error message" />
+      </Stack>
+
+      <Stack gap="15" styles={{ root: { width: 300 } }}>
+        <MaskedTextField label="With input mask" mask="m\ask: (999) 999 - 9999" />
+
```
Comment (khmakoto) [inline packages/office-ui-fabric-react/src/components/TextField/examples/TextField.Basic.Example.tsx]: nit: Why are there blank lines separating these? (Same with other files below)

Comment (ecraig12345) [reply packages/office-ui-fabric-react/src/components/TextField/examples/TextField.Basic.Example.tsx]: Attempt at improving readability...not sure if it actually helps

**Pull request merged**